src/main/java/org/broadinstitute/hellbender/tools/coveragemodel/CoverageModelEMComputeBlock.java

```diff
@@ -0,0 +1,1332 @@
+                /* mask */
+                .addPrimitiveNode(CoverageModelICGCacheNode.M_st.name(),
+                        new String[]{},
+                        new DuplicableNDArray())
+                /* mapping error rate */
+                .addPrimitiveNode(CoverageModelICGCacheNode.MER_st.name(),
+                        new String[]{},
+                        new DuplicableNDArray());
+
+        /*
+         * Model parameters
+         */
+        cgbuilder
+                /* mean log bias */
+                .addPrimitiveNode(CoverageModelICGCacheNode.m_t.name(),
+                        new String[]{},
+                        new DuplicableNDArray())
+                /* unexplained variance */
+                .addPrimitiveNode(CoverageModelICGCacheNode.Psi_t.name(),
+                        new String[]{},
+                        new DuplicableNDArray())
+                /* model bias covariates */
+                .addPrimitiveNode(CoverageModelICGCacheNode.W_tl.name(),
+                        new String[]{},
+                        new DuplicableNDArray());
+
+        /*
+         * Externally determined computable nodes (all of latent posterior expectations + etc)
+         */
+        cgbuilder
+                /* E[log(c_{st})] */
+                .addComputableNode(CoverageModelICGCacheNode.log_c_st.name(),
+                        new String[]{},
+                        new String[]{},
+                        null, true)
+                /* var[log(c_{st})] */
+                .addComputableNode(CoverageModelICGCacheNode.var_log_c_st.name(),
+                        new String[]{},
+                        new String[]{},
+                        null, true)
+                /* E[log(d_s)] */
+                .addComputableNode(CoverageModelICGCacheNode.log_d_s.name(),
+                        new String[]{},
+                        new String[]{},
+                        null, true)
+                /* var[log(d_s)] */
+                .addComputableNode(CoverageModelICGCacheNode.var_log_d_s.name(),
+                        new String[]{},
+                        new String[]{},
+                        null, true)
+                /* E[\gamma_s] */
+                .addComputableNode(CoverageModelICGCacheNode.gamma_s.name(),
+                        new String[]{},
+                        new String[]{},
+                        null, true)
+                /* E[z_{sm}] */
+                .addComputableNode(CoverageModelICGCacheNode.z_sl.name(),
+                        new String[]{},
+                        new String[]{},
+                        null, true)
+                /* E[z_{sm} z_{sn}] */
+                .addComputableNode(CoverageModelICGCacheNode.zz_sll.name(),
+                        new String[]{},
+                        new String[]{},
+                        null, true)
+                /* FFT[W] */
+                .addComputableNode(CoverageModelICGCacheNode.F_W_tl.name(),
```

Comment:
This is commented out for now. To answer your question -- it is externally computed, and externally computed nodes may have parents in general. Also, there is a level of ad-hoc-ness to how we define the DAG. In theory, representing variational inference in a computation graph does not yield a DAG at all -- it clearly has recurrence. What we do here is somewhat hacky: we break the cycles via orphan externally computable nodes and let the driver handle fixed-point iterations. In a sense, externally computable nodes mark the finishing line of a fixed-point iteration cycle.

If we were to write a _proper_ framework for VI, we should have let ICG represent a DCG. Each computable node would have an _iteration number_ attached to it, and it would go out-of-date only if it were behind the querying node by a certain number of iterations. Here, we are assuming that some external driver is taking care of the order of updating various nodes and use ICG basically as a smart cache that performs some of the computations for us.